chainer/functions/connection/deconvolution_2d.py

```diff
@@ -0,0 +1,300 @@
+        ksize = _pair(ksize)
+        stride = _pair(stride)
+        pad = _pair(pad)
+
+        self.kh, self.kw = ksize
+        self.sy, self.sx = stride
+        self.ph, self.pw = pad
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+
+        self.W = None
+        self.gW = None
+        self.b = None
+        self.gb = None
+
+        if initialW is not None:
+            assert initialW.shape == \
+                (in_channels, out_channels, self.kh, self.kw)
+            self.W = initialW
+        else:
+            self.W = numpy.random.normal(
+                0, wscale * math.sqrt(1. / (self.kh * self.kw * in_channels)),
+                (in_channels, out_channels, self.kh, self.kw)
+            ).astype(numpy.float32)
+        xp = cuda.get_array_module(self.W)
+        self.gW = xp.full_like(self.W, numpy.nan)
+
+        if initial_bias is not None:
+            assert initial_bias.shape == (out_channels,)
+            self.b = initial_bias
+        elif not nobias:
+            self.b = numpy.repeat(numpy.float32(bias), out_channels)
+
+        if self.b is not None:
+            self.gb = xp.full_like(self.b, numpy.nan)
+
+        self.use_cudnn = use_cudnn
+        # chance to choose implicit-precomp-gemm algorithm
+        self.max_workspace_size = out_channels * self.kh * self.kw * 4
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+
+        type_check.expect(
+            x_type.dtype == numpy.float32,
+            x_type.ndim == 4,
+            x_type.shape[1] == self.in_channels
+        )
+
+    def check_type_backward(self, in_types, out_types):
+        type_check.expect(out_types.size() == 1)
+        x_type, = in_types
+        y_type, = out_types
+
+        in_h = x_type.shape[2].eval()
+        out_h = type_check.Variable(
+            get_deconv_outsize(in_h, self.kh, self.sy, self.ph),
+            'out_h')
+        in_w = x_type.shape[3].eval()
+        out_w = type_check.Variable(
+            get_deconv_outsize(in_w, self.kw, self.sx, self.pw),
+            'out_w')
+
+        type_check.expect(
+            y_type.dtype == numpy.float32,
+            y_type.ndim == 4,
+            y_type.shape[0] == x_type.shape[0],
+            y_type.shape[1] == self.out_channels,
+            y_type.shape[2] == out_h,
+            y_type.shape[3] == out_w
+        )
+
+    @property
+    def parameter_names(self):
+        if self.b is None:
+            return 'W',
+        return 'W', 'b'
+
+    @property
+    def gradient_names(self):
+        if self.gb is None:
+            return 'gW',
+        return 'gW', 'gb'
+
+    def zero_grads(self):
+        self.gW.fill(0)
+        if self.gb is not None:
+            self.gb.fill(0)
+
+    def forward_cpu(self, x):
+        n, c, h, w = x[0].shape
+        gcol = numpy.tensordot(self.W, x[0], (0, 1))
+        # k, m, n, b, h, w
```

Comment:
I would like to add information that `k`, `m`, `n` indicate the # of output channels(`out_channel`), the height of kernels `kh` and the width of kernels `kw`, respectively.